Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -115,10 +138,24 @@ public sealed class AddIshBackgroundTask : BackgroundTaskCmdlet
         [AllowEmptyCollection]
         public IshObject[] IshObject { get; set; }
 
+        /// <summary>
+        /// <para type="description">The InputDataTemplate (e.g. IshObjectWithLngRef) indicates whether a list of ishObjects or one ishObject is submitted as input data to the background task.</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectsGroup")]
+        public InputDataTemplate InputDataTemplate { get; set; } // Change to enum.
+
+        /// <summary>
+        /// <para type="description">The hash id of the background task.</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectsGroup")]
+        [ValidateLength(0, 80)]
+        public string HashId { get; set; }
+
         #region Private fields
         private readonly List<IshObject> _retrievedIshObjects = new List<IshObject>();
         private readonly DateTime _modifiedSince = DateTime.Today.AddDays(-1);
-        private readonly BackgroundTask25ServiceReference.eUserFilter _userFilter = EnumConverter.ToUserFilter<BackgroundTask25ServiceReference.eUserFilter>(Enumerations.UserFilter.Current);
+        private readonly eUserFilter _userFilter = EnumConverter.ToUserFilter<eUserFilter>(Enumerations.UserFilter.Current);
```

Comment:
I expected an `_inputDataTemplate` here, like this `EnumConverter.ToUserFi...` line. Because you made `InputDataTemplate` a non-mandatory parameter, so if not specified it defaults to...? 
Add a default for `IshObjectsWithLngRef ` because any customer that already has a script that reads `Add-IshBackgroundTask -EventType "SMARTTAG"` should not break